detection_rules/schemas/definitions.py

```diff
@@ -44,24 +44,31 @@
 }
 
 
+def string_only_list(obj: List[str]):
+    """Validate all records in a list are strings."""
+    return all([isinstance(r, str) for r in obj])
+
+
 BranchVer = NewType('BranchVer', str, validate=validate.Regexp(BRANCH_PATTERN))
 CodeString = NewType("CodeString", str)
 ConditionSemVer = NewType('ConditionSemVer', str, validate=validate.Regexp(CONDITION_VERSION_PATTERN))
 Date = NewType('Date', str, validate=validate.Regexp(DATE_PATTERN))
 FilterLanguages = Literal["kuery", "lucene"]
 Interval = NewType('Interval', str, validate=validate.Regexp(INTERVAL_PATTERN))
-PositiveInteger = NewType('PositiveInteger', int, validate=validate.Range(min=1))
 Markdown = NewType("MarkdownField", CodeString)
 Maturity = Literal['development', 'experimental', 'beta', 'production', 'deprecated']
 MaxSignals = NewType("MaxSignals", int, validate=validate.Range(min=1))
 NonEmptyStr = NewType('NonEmptyStr', str, validate=validate.Length(min=1))
 Operator = Literal['equals']
 OSType = Literal['windows', 'linux', 'macos']
+PositiveInteger = NewType('PositiveInteger', int, validate=validate.Range(min=1))
 RiskScore = NewType("MaxSignals", int, validate=validate.Range(min=1, max=100))
 RuleType = Literal['query', 'saved_query', 'machine_learning', 'eql', 'threshold', 'threat_match']
 SemVer = NewType('SemVer', str, validate=validate.Regexp(VERSION_PATTERN))
 Severity = Literal['low', 'medium', 'high', 'critical']
 Sha256 = NewType('Sha256', str, validate=validate.Regexp(SHA256_PATTERN))
+StringListMaxSizeThree = NewType('StringListMaxSizeThree', list,
+                                 validate=validate.And(validate.Length(min=0, max=3), string_only_list))
```

Comment:
since this use case is so tiny, what if we just call it `Cardinality`?
Also, any objection to this change? should it be `List[str]` or `List[NonEmptyStr]`?
```suggestion
StringListMaxSizeThree = NewType('StringListMaxSizeThree', List[NonEmptyStr],
                                 validate=validate.Length(min=0, max=3), 
```

Reply:
Normally I would agree, but since we are limiting all of our field validations to occur within the type definitions, this will not be the only time we have to make a type with validation that may be overly specific like this. So in these cases, it feels like descriptiveness, though longer (and maybe more awkward) are actually more favorable.

If you feel strongly, I wouldn't fight hard, but I think it is ok to maintain naming conventions like this going forward, especially since it is in a dedicated `definitions` file.

Good call on the `NonEmptyStr` though 👍

Reply:
i think we've already gone down that road and it's okay. like we did for FilterLanguages.

Reply:
True, but `FilterLanguages` makes sense since it is a descriptive literal which can be defined specifically pretty easily.

A list with max length is generic enough that it _can_ be applied elsewhere (or even  `StringListMaxSizeFour`, etc.).

But like I said, no strong feelings, so if you still insist (via one more comment), I will oblige 🙂